source/postprocess/visualization/boundary_indicator.cc

```diff
@@ -0,0 +1,98 @@
+/*
+  Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/visualization/boundary_indicator.h>
+#include <aspect/simulator_access.h>
+
+
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      template <int dim>
+      std::pair<std::string, Vector<float> *>
+      BoundaryIndicator<dim>::execute() const
+      {
+        std::pair<std::string, Vector<float> *>
+        return_value ("boundary_indicator",
+                      new Vector<float>(this->get_triangulation().n_active_cells()));
+
+        // loop over all of the surface cells and extract boundary indicators
+        typename DoFHandler<dim>::active_cell_iterator
+        cell = this->get_dof_handler().begin_active(),
+        endc = this->get_dof_handler().end();
+
+        unsigned int cell_index = 0;
+        for (; cell!=endc; ++cell,++cell_index)
+          if (cell->is_locally_owned())
+            if (cell->at_boundary())
+              {
+                types::boundary_id boundary_id;
+                for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+                  {
+                    boundary_id = cell->face(f)->boundary_indicator();
+                    if (boundary_id != numbers::internal_face_boundary_id)
+                      break;
+                  }
+                (*return_value.second)(cell_index) = static_cast<unsigned int> (boundary_id);
```

Comment:
I think you want to cast this to `float`, not `unsigned int`, since that's what you have on the left hand side.